lib/galaxy/objectstore/__init__.py

```diff
@@ -509,89 +639,108 @@ def __filesystem_monitor(self):
+                    obj.object_store_id = random.choice(
+                            self.weighted_backend_ids)
                 except IndexError:
-                    raise ObjectInvalid( 'objectstore.create, could not generate obj.object_store_id: %s, kwargs: %s' % ( str( obj ), str( kwargs ) ) )
-                create_object_in_session( obj )
-                log.debug("Selected backend '%s' for creation of %s %s" % (obj.object_store_id, obj.__class__.__name__, obj.id))
+                    raise ObjectInvalid(
+                            'objectstore.create, could not generate '
+                            'obj.object_store_id: %s, kwargs: %s'
+                            % ( str( obj ), str( kwargs ) ) )
+                _create_object_in_session( obj )
+                log.debug("Selected backend '%s' for creation of %s %s"
+                        % (obj.object_store_id,
+                            obj.__class__.__name__, obj.id))
             else:
-                log.debug("Using preferred backend '%s' for creation of %s %s" % (obj.object_store_id, obj.__class__.__name__, obj.id))
+                log.debug("Using preferred backend '%s' for creation of %s %s"
+                        % (obj.object_store_id,
+                            obj.__class__.__name__, obj.id))
             self.backends[obj.object_store_id].create(obj, **kwargs)
 
-    def _call_method(self, method, obj, default, default_is_exception, **kwargs):
+    def _call_method(self, method, obj, default, default_is_exception,
+            **kwargs):
         object_store_id = self.__get_store_id_for(obj, **kwargs)
         if object_store_id is not None:
-            return self.backends[object_store_id].__getattribute__(method)(obj, **kwargs)
+            return self.backends[object_store_id].__getattribute__(method)(obj,
+                    **kwargs)
         if default_is_exception:
-            raise default( 'objectstore, _call_method failed: %s on %s, kwargs: %s' % ( method, str( obj ), str( kwargs ) ) )
+            raise default(
+                    'objectstore, _call_method failed: %s on %s, kwargs: %s'
+                    % ( method, str( obj ), str( kwargs ) ) )
         else:
             return default
 
     def __get_store_id_for(self, obj, **kwargs):
-        if obj.object_store_id is not None and obj.object_store_id in self.backends:
+        if obj.object_store_id is not None and \
+                obj.object_store_id in self.backends:
             return obj.object_store_id
         else:
             # if this instance has been switched from a non-distributed to a
             # distributed object store, or if the object's store id is invalid,
             # try to locate the object
-            log.warning('The backend object store ID (%s) for %s object with ID %s is invalid' % (obj.object_store_id, obj.__class__.__name__, obj.id))
+            log.warning(
+                    'The backend object store ID (%s) for %s object with ID "
```

Comment:
While I do like Makefiles, `tox` executed from the galaxy root dir currently should perform the function of running the flake8'ing (along with verifying unit tests) in a standard way so that someone can validate prior to submitting a PR.  I have to say I'm not a fan of anyone autopep8'ing anything ;)